types/react-router-native/index.d.ts

```diff
@@ -31,15 +31,24 @@ export class AndroidBackButton extends React.Component<AndroidBackButtonProps> {
 export class DeepLinking extends React.Component {}
 
 export interface LinkProps {
-  component?: React.Component;
+  component?: React.Component | React.ComponentClass;
   replace?: boolean;
   style?: any;
   to: H.LocationDescriptor;
 }
 
-export class Link extends React.Component<LinkProps> {}
+/**
+ * Disable the "no-any-union" rule here as the Link component can accept any property.
+ * It will feed them to the given component. However if we just use any we will lose
+ * autocomplete on known properties. So it is better to disable this rule in this one
```

Comment:
First off

![image](https://user-images.githubusercontent.com/972891/36873650-f24d77a2-1d5d-11e8-977e-bc91f9aea779.png)

I'm not seeing any augmentation in completion. Second, if `LinkProps` can accept any property, you'd be better off adding an index signature like

```diff
   replace?: boolean;
   style?: any;
   to: H.LocationDescriptor;
+  [propName: string]: any;
 }
```

Reply:
as seems i missed that inside the common mistake section, sorry about that will fix